components/src/dynamo/vllm/handlers.py

```
@@ -1474,14 +1474,73 @@ def __init__(self, d: dict) -> None:
+                )
+                if new_dp_size == prev_dp:
+                    # No size change was applied; nothing to roll back.
+                    raise
+                # Roll back to the last good dp: "either it grows, or nothing
+                # changes" for the engine, not just the pod.
+                try:
+                    logger.warning(
+                        f"[ElasticEP] Rolling back to data_parallel_size={prev_dp}"
+                    )
+                    await _scale_engine(prev_dp)
+                except Exception as rollback_err:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)